samples/go/json-import/json_importer.go

```diff
@@ -8,16 +8,68 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
 	"log"
+	"math/big"
 	"net/http"
 	"os"
+	"time"
 
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/spec"
 	"github.com/attic-labs/noms/go/util/jsontonoms"
+	"github.com/dustin/go-humanize"
 	flag "github.com/juju/gnuflag"
 )
 
+func newProgressReader(r io.Reader) io.Reader {
+	duration := time.Second / 10
+	timer := time.NewTimer(duration)
+	progress := make(chan int)
+	done := make(chan bool)
+	go func() {
+		start := time.Now()
+		completed := int64(0)
+		reportProgress := func() {
+			elapsed := time.Since(start)
+			rate := float64(completed) * (float64(time.Second) / float64(elapsed))
+			clearLine := "\x1b[2K\r"
```

Comment:
We have this factored into a utility package you can use at `github.com/attic-labs/noms/go/util/status`.